src/containers/Petitions.js

```diff
@@ -1,26 +1,76 @@
 import React from 'react';
 import Helmet from 'react-helmet';
 import { connect } from 'react-redux';
-import { fetchPetitions } from 'actions/PetitionsActions';
+import { withRouter } from 'react-router';
+import { petitionsPath } from 'helpers/petitionUrls';
+import { fetchPetitions, updateCityFilterValue } from 'actions/PetitionsActions';
 import settings from 'settings';
 import Petitions from 'components/Petitions';
 import getPetitions from 'selectors/petitions';
 
-const PetitionsContainer = React.createClass({
+const PetitionsContainer = withRouter(React.createClass({
   componentWillMount () {
     this.props.fetchPetitions(this.props);
   },
 
+  componentWillReceiveProps (nextProps) {
+    if (this.props.location.pathname !== nextProps.location.pathname) {
+      this.props.fetchPetitions(nextProps);
+    }
+  },
```

Comment:
I am working on this in the pagination branch. The idea is, all params / query params are stored in the state e.g.

```
{
  city: 'xx',
  page: 2,
  limit: 50
}
```

Then we deep compare whether any params have changed, then call `this.props.fetchPetitions` if so